### ISSUE TITLE
docs - createCookieSessionStorage says commitSession not needed

### ISSUE THREAD

**Issue by penx** [remix-run/remix#9444]
The docs for `createCookieSessionStorage` state:

> With other session storage strategies you only have to commit it when it's created

https://remix.run/docs/en/main/utils/sessions#createcookiesessionstorage

When we moved from `createCookieSessionStorage` to another storage solution (vercel kv), we followed this advice and removed calls to `commitSession` (assuming it was not needed, and that `session.set` would be all that is required) but it resulted in sessions not being saved.

Elsewhere in the docs for `createSessionStorage`:

> updateData will be called from commitSession

https://remix.run/docs/en/main/utils/sessions#createsessionstorage

Which seems to imply that anything based on `createSessionStorage` will require a call to `commitSession` in order to update the session, and I'm fairly sure explains why we would have bugs when removing calls to `commitSession`.

I think the docs for `createCookieSessionStorage` should instead state

> With other session storage strategies you only have to _send an updated cookie in the request header_ when it's created

I'll raise a PR to the docs for this.

**Comment by kiliman** [comment]
Yes, the documentation needs to be corrected. For the session API, all session storage providers require you to call `commitSession` after mutation.

I think what the documentation was alluding to was you don't have to return a `set-cookie` header in the response after every commit, just the initial one.

In cookie session storage, the session data is stored entirely in the cookie, so any changes require you to send an updated cookie via the `set-cookie` header.

However, in other storage providers like Cloudflare KV or file-based sessions, the only thing stored in the cookie is the session ID. You would send that id in the `set-cookie` header on creation. The session ID is used to look up the session in the storage. Since the session id never changes, you don't need to keep sending a `set-cookie` header.

**Comment by akomm** [comment]
Might be related with this issue?
https://github.com/remix-run/remix/discussions/9338

**Comment by brophdawg11** [comment]
Thank you for opening this issue, and our apologies we haven't gotten around to it yet!

With the release of [React Router v7](https://remix.run/blog/react-router-v7) we are sun-setting continued development/maintenance on Remix v2. If you have not already [upgraded to React Router v7](https://reactrouter.com/upgrading/remix), we recommend you do so. We've tried to make the upgrade process as smooth as possible with our [Future Flags](https://remix.run/docs/en/main/start/future-flags). We are now in the process of cleaning up outdated issues and pull requests to improve the overall hygiene of our repositories.

We plan to continue to address 2 types of issues in Remix v2:

- Bugs that pose security concerns
- Bugs that prevent upgrading to React Router v7

If you believe this issue meets one of those criteria, please respond or [create a new issue](https://github.com/remix-run/remix/issues/new?template=bug_report.yml).

For all other issues, ongoing maintenance will be happening in React Router v7, so:

- If this is a bug, please reopen this issue [in that repo](https://github.com/remix-run/react-router/issues/new?template=bug_report.yml) with a new [minimal reproduction](https://reactrouter.com/new) against v7
- If this is a feature request, please open a new [Proposal Discussion](https://github.com/remix-run/react-router/discussions/new?category=proposals) in React Router, and if it gets enough community support it can be considered for implementation

If you have any questions you can always reach out on [Discord](https://rmx.as/discord). Thanks again for providing feedback and helping us make our framework even better!
